### PR TITLE
Stop Item Add Duplicate

### LINKED ISSUE #179: Clicking on the Add Item Button twice creates duplicate
**Describe the bug**

When users go to add a vocab item, it may take some time. Users may not know this, and click the button twice or thrice. This creates duplicates of the same vocab item. We should disable buttons after they are clicked, and should only be enabled after the `onPress` function for that button has been resolved.

### PR DESCRIPTION
**Pull request by eeshabarua**
## Description
WE FINALLY GOT IT we set a useState boolean to disable the button, setting isDisable while the success callback is running and false upon success.

Fixes #179

### CODE REVIEW

File: client/src/pages/VocabDrawer/VocabDrawer.js [hunk 1]
```diff
@@ -170,112 +170,118 @@ const VocabDrawer = ({ navigation }) => {
     navigation.goBack()
   }
 
+  const [isDisabled, setIsDisabled] = useState(false)
+
   /**
    * Either updates the vocab item or creates a new vocab item
    */
```
Comment (ashayp22) [inline client/src/pages/VocabDrawer/VocabDrawer.js:177]: Amazing! Were you able to test this? Also, where is the state variable `isDisabled` actually being checked?

Comment (eeshabarua) [reply client/src/pages/VocabDrawer/VocabDrawer.js:177]: Yes, it works in testing! Also, it is being checked and updated accordingly in the success callback.

File: client/src/pages/VocabDrawer/VocabDrawer.js [hunk 2]
```diff
@@ -170,112 +170,118 @@ const VocabDrawer = ({ navigation }) => {
...
-      },
-      () => {
-        close() // on success, close the modal
-      },
-    )
+        },
+        () => {
+          close() // on success, close the modal
+          setIsDisabled(false)
+        },
```
Comment (ashayp22) [inline client/src/pages/VocabDrawer/VocabDrawer.js:282]: change drawer and add IsDisabled in drawer instead to edit vocab, unit and lesson

**Review by ashayp22 — CHANGES_REQUESTED** [review]
change the functionality from vocab drawer to drawer

**Comment by ashayp22** [comment]
#225 deals with the same issue.

**Pull request closed without merging**